pkg/registry/core/service/ipallocator/allocator_test.go

```diff
@@ -163,18 +163,38 @@ func TestAllocateSmall(t *testing.T) {
 }
 
 func TestRangeSize(t *testing.T) {
-	testCases := map[string]int64{
-		"192.168.1.0/24": 256,
-		"192.168.1.0/32": 1,
-		"192.168.1.0/31": 2,
+	testCases := []struct {
+		cidr  string
+		addrs int64
+	}{
+		{
+			cidr:  "192.168.1.0/24",
+			addrs: 256,
+		}, {
+			cidr:  "192.168.1.0/32",
+			addrs: 1,
+		}, {
+			cidr:  "192.168.1.0/31",
+			addrs: 2,
+		}, {
+			cidr:  "2001:db8::/98",
+			addrs: 1073741824,
+		}, {
+			cidr:  "2001:db8::/128",
+			addrs: 1,
+		}, {
+			cidr:  "2001:db8::/127",
+			addrs: 2,
+		},
```

Comment:
Agreed. I had fail cases in my original commit but I then removed them. As you mention in your other comment, I think the func needs to be updated to return an error.